src/formatDate.js

```diff
@@ -9,6 +9,54 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+
+  const oldDateParts = date.split(oldSeparator);
+
+  let datePartsMatched = {};
+
+  for (let i = 0; i < oldDateParts.length; i++) {
+    datePartsMatched[fromFormat[i]] = oldDateParts[i];
+  }
+
+  datePartsMatched = convertYear(datePartsMatched);
```

Comment:
The `convertYear` function is being called with the `datePartsMatched` object, but the function itself is not defined until later in the code. It's a good practice to define functions before they are used to improve readability and avoid hoisting issues. Consider moving the `convertYear` function definition above its first usage.